include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -191,8 +191,9 @@ class argument_parser
                                "(regex: \"^[a-zA-Z0-9_-]+$\")."};
 
         for (auto & sub : this->subcommands)
-            if (!std::regex_match(sub, std::regex{"^[a-zA-Z0-9_]+$"}))
-                throw design_error{"The subcommand name must only contain alpha-numeric characters or '_'."};
+            if (!std::regex_match(sub, app_name_regex))
+                throw design_error{"The subcommand name must only contain alpha-numeric characters or '_' and '-' "
+                                   "(regex: \"^[a-zA-Z0-9_-]+$\")."};
```

Comment:
Since it now has multiple lines, we should add braces .. can you also do it for the other throw above?